helpers/lib/common.js

```diff
@@ -24,10 +24,20 @@ function isTruthy(val) {
     }
 }
 
+function matchesPredicate(element, predicate, predicateKeys) {
```

Comment:
I would change this so the caller doesn't have to pass in `predicateKeys`. That should be internal to this function.

Reply:
I was thinking that it would be expensive to keep calling Object.keys(predicate) for each element.

Reply:
But will change it.